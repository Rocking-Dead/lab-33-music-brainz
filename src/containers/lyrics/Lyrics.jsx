import React from 'react';
import PropTypes from 'prop-types';

const Lyrics = ({lyrics}) => {
  const parsedLyrics = lyrics.split("\n")
  .map((newLine) => {
    return 
    <li key={id}>{newLine}</li>
  });

  return <ul data-testid="Lyrics">{parsedLyrics}</ul>;
};

Lyrics.propTypes = {
    lyrics: PropTypes.string.isRequired,
};

export default Lyrics;

