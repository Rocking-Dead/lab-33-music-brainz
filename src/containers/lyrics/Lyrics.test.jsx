import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Lyrics from '../lyrics/Lyrics';

describe('Lyrics component', () => {
  afterEach(() => cleanup());
  it('renders Lyrics', () => {
    const { asFragment } = render(<Lyrics lyrics={''} />);
    expect(asFragment()).toMatchSnapshot();
  });
});


