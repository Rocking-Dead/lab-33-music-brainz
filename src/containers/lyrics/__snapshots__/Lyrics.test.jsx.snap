// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics component renders Lyrics 1`] = `
<DocumentFragment>
  <ul
    data-testid="Lyrics"
  />
</DocumentFragment>
`;
